### PR TITLE
Repo restructure

### PR DESCRIPTION
**Pull request by jmckenna**
- related to plan mentioned through https://github.com/MapServer/MapServer/issues/6837
- resulting tree visible at https://github.com/jmckenna/MapServer/tree/repo-restructure-8.2-release

## Layout

```
MapServer/
      .github/
      ci/
      cmake/
      etc/
      fonts/
      fuzzers/
      msautotest/
      scripts/
      share/
      src/
          apache/
          apps/
            coshp.*
            legend.*
            map2img.*
            mapserv.*
            msencrypt.*
            scalebar.*
            shptree*.*
            sortshp.*
            tile4ms.*
          flatgeobuf/
          mapscript/
          opengl/
          renderers/
          third-party/
          xmlmapfile/
          *.c
          *.cpp
          *.h
      symbols/
      tests/
      .travis.yml
      appveyor.yml
      CMakeLists.txt
      CONTRIBUTING.md
      HISTORY.md
      LICENSE.md
      README.md
      SECURITY.md
```

### CODE REVIEW

**Comment by jmckenna** [comment]
@sdlime maybe another set of eyes can help me here, as only 1 `msautotest` fails, this one : https://github.com/jmckenna/MapServer/blob/main/msautotest/config/hello_world.map#L2

```
# RUN_PARMS: hello_world.png [MAPSERV] "PATH_INFO=/[MAPFILE]" QUERY_STRING="mode=map" > [RESULT_DEMIME]
```
An empty/invalid PNG is created.  I can reproduce this locally.

I wonder what is causing this....

**Comment by sdlime** [comment]
All the other config-related tests run ok, correct?

**Comment by sdlime** [comment]
> correct, only the one `config` run mentioned here fails (the second one in `hello_world.map`).

Looking at hello_world.map it seems like those first 3 tests are set up incorrectly.

```
# RUN_PARMS: hello_world.png [MAPSERV] QUERY_STRING="map=[MAPFILE]&mode=map" > [RESULT_DEMIME]
# RUN_PARMS: hello_world.png [MAPSERV] "PATH_INFO=/[MAPFILE]" QUERY_STRING="mode=map" > [RESULT_DEMIME]
# RUN_PARMS: hello_world.png [MAPSERV] "PATH_INFO=/[MAPFILE]/" QUERY_STRING="mode=map" > [RESULT_DEMIME]
```

They really should write 3 separate output files - hello_world1.png, hello_world2.png, etc... - or more descriptive names? @geographika might have suggestions. I just wonder if those tests are stepping on one another somehow and causing problems.

--Steve

**Comment by jmckenna** [comment]
Yes I had wondered that also, and had made them generate `hello_world.png1` for the first one, with an expected image of `hello_world1.png`, and so on for the 3 tests, however that did not solve the issue with the second run.  So the stepping-on theory is not valid.   (I have been battling this a few days).

**Comment by jmckenna** [comment]
@sdlime key is executing that second run locally, to see what I mean (I deleted all of the other runs in `hello_world.map` locally, to test).   You can run `.github/workflows/start.sh` locally to get to this same spot.

**Comment by jmckenna** [comment]
@sdlime ah good idea, yes I understand, will-do.  (boy this is a tricky one)

**Comment by sdlime** [comment]
Yeah, the _expected_ file in the repo is from the 3rd call (I think).

**Comment by geographika** [comment]
The second test is to make sure that a Mapfile key in a CONFIG file can be accessed via a URL as per https://github.com/MapServer/MapServer/commit/497ba5d3c24ce5bc64df0cd3e78dbfce8308d8b9 
A valid PNG identical to the first test should be returned. If the third test works then there must be an issue with slashes in PATH_INFO. 
Unfortunately I won't be able to test for a few days.

**Comment by geographika** [comment]
@jmckenna - is there any difference with the following test?

```
# RUN_PARMS: hello_world_mapfile_url.png [MAPSERV] -conf ms_no_post.conf "PATH_INFO=/HELLO_WORLD" QUERY_STRING="mode=map" > [RESULT_DEMIME]
```

**Comment by jmckenna** [comment]
@geographika unfortunately it generates the same result, a 172B invalid PNG file.

**Comment by jmckenna** [comment]
thanks @sdlime the other `hello_world_mapfile` images are 2.2KB

When I try opening the image in a text editor it shows:
```
<HTML>
<HEAD><TITLE>MapServer Message</TITLE></HEAD>
<BODY BGCOLOR="#FFFFFF">
msCGILoadMap(): Web application error. CGI variable &quot;map&quot; is not set.
</BODY></HTML>
```

File: CMakeLists.txt [hunk 1]
```diff
@@ -307,41 +307,41 @@ else(WIN32)
...
+src/mapflatgeobuf.h src/mapgraph.h src/mapserv-config.h)
 
 if(WIN32)
 	configure_file(
-	  ${CMAKE_CURRENT_SOURCE_DIR}/version.rc.in
-	  ${CMAKE_CURRENT_BINARY_DIR}/version.rc
+	  ${CMAKE_CURRENT_SOURCE_DIR}/src/version.rc.in
+	  ${CMAKE_CURRENT_BINARY_DIR}/src/version.rc
 	  @ONLY)
 	set(mapserver_SOURCES ${mapserver_SOURCES} ${CMAKE_CURRENT_BINARY_DIR}/version.rc)
```
Comment (geographika) [inline CMakeLists.txt:346]: @jmckenna - I think this also needs to be updated to:

`set(mapserver_SOURCES ${mapserver_SOURCES} ${CMAKE_CURRENT_BINARY_DIR}/src/version.rc)`

See errors in https://ci.appveyor.com/project/MapServer/mapserver/builds/47765494/job/fw9usno6l5315808

Comment (jmckenna) [reply CMakeLists.txt:346]: good catch, thanks @geographika

**Comment by geographika** [comment]
@jmckenna - the following line in the appveyor.yml build config will need to be updated:

https://github.com/MapServer/MapServer/blob/58ee7031e34d2445c9f6f4c0591ab85845dd8063/appveyor.yml#L98

It should be:

```
- "%Python_ROOT_DIR%/python -m pip install --no-index --find-links=file://%BUILD_FOLDER%/build/src/mapscript/python/Release/dist mapscript"
```

**Comment by geographika** [comment]
The Appveyor Windows builds are now working. I can't see what the errors are for the GHA and Travis builds though beyond that cryptic `[Makefile:85: test] Error 2`. All the tests seem to pass.

**Comment by jmckenna** [comment]
@geographika yes agreed, will examine this locally...

**Comment by rouault** [comment]
Fixes for the remaining errors in https://github.com/jmckenna/MapServer/pull/5

**Comment by geographika** [comment]
@jmckenna - could the vagrant file also require changes with the new structure?
https://github.com/MapServer/MapServer/blob/main/scripts/vagrant/mapserver.sh

**Comment by jmckenna** [comment]
> @jmckenna - could the vagrant file also require changes with the new structure? https://github.com/MapServer/MapServer/blob/main/scripts/vagrant/mapserver.sh

thanks @geographika, tackled through https://github.com/MapServer/MapServer/pull/6919/commits/2d4543aedaaad76c80664a6d5a9236bc19868e18

**Comment by jmckenna** [comment]
@sdlime I think this is now ready for your review also.

**Comment by jmckenna** [comment]
Will give until Friday for additional comments, and then will merge.

**Comment by rouault** [comment]
> and then will merge.

when merging, please use "Squash and merge" (through the drop down combobox of the "Create a merge commit" green button) to generate a single commit, so that "git bisect" sessions work properly (otherwise they could it intermediate commits that don't build)

**Comment by jmckenna** [comment]
thanks @rouault will-do

**Comment by jmckenna** [comment]
PS. I know you so well, that after I wrote that I had thought "should I edit my comment and change my wording to squash?", ha, small world.  My apologies for my wording.

**Comment by sdlime** [comment]
Things do look a lot more tidy - thanks for enduring... I approve.

**Pull request merged**